jobs: NPE in TestFullClusterBackup

Failing build: https://teamcity.cockroachdb.com/viewLog.html?buildId=3489692&buildTypeId=Cockroach_UnitTests_Test&tab=artifacts#%2FTestFullClusterBackup628874167%2Fthrowawayjob;%2FTestFullClusterBackup628874167%2Ffoo;%2Fraw.0.json.txt.tgz

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x58 pc=0x21595f5]

goroutine 754544 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc018ab8780, 0x62a8828, 0xc006ade600)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:231 +0x126
panic(0x458a560, 0x83aabd0)
        /usr/local/go/src/runtime/panic.go:965 +0x1b9
github.com/cockroachdb/cockroach/pkg/jobs.(*ScheduledJob).SetScheduleStatus(0x0, 0x4c64327, 0x7, 0x0, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/jobs/scheduled_job.go:242 +0x35
github.com/cockroachdb/cockroach/pkg/sql.(*sqlStatsCompactionResumer).Resume.func1(0x62a88d0, 0xc013ec31a0, 0xc000f61760, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/compact_sql_stats.go:60 +0x1ce
github.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn.func1(0x62a88d0, 0xc013ec31a0, 0xc000f61760, 0xc003c28900, 0x3ff0000000000000)
        /go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:818 +0x43
github.com/cockroachdb/cockroach/pkg/kv.(*Txn).exec(0xc000f61760, 0x62a88d0, 0xc013ec31a0, 0xc0157b55d8, 0xc000f61760, 0x2a)
        /go/src/github.com/cockroachdb/cockroach/pkg/kv/txn.go:890 +0xdc
github.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn(0xc018ab9680, 0x62a88d0, 0xc013ec31a0, 0xc0157b5690, 0xc008ead870, 0x21)
        /go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:817 +0x105
github.com/cockroachdb/cockroach/pkg/sql.(*sqlStatsCompactionResumer).Resume(0xc014f0b7d0, 0x62a88d0, 0xc013ec31a0, 0x496c020, 0xc008ead870, 0x4587500, 0xc0014f7740)
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/compact_sql_stats.go:52 +0x1ac
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine.func2(0xc008e57ec0, 0x62634c0, 0xc014f0b7d0, 0x62a88d0, 0xc013ec31a0, 0x496c020, 0xc008ead870, 0xc0157b58e0)
        /go/src/github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1180 +0xae
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine(0xc009591c00, 0x62a88d0, 0xc013ec3110, 0x496c020, 0xc008ead870, 0x62634c0, 0xc014f0b7d0, 0xc014cc6840, 0xc00ac14b98, 0x7, ...)
        /go/src/github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1181 +0x907
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).runJob(0xc009591c00, 0x62a8828, 0xc01151bfc0, 0x62634c0, 0xc014f0b7d0, 0xc014cc6840, 0xc00ac14b98, 0x7, 0xc0083b2e58, 0x16, ...)
        /go/src/github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:415 +0x479
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resumeJob.func1(0x62a8828, 0xc006ade600)
        /go/src/github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:336 +0x165
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2(0xc018ab8780, 0x62a8828, 0xc006ade600, 0x0, 0x0, 0xc012cefcc0)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:442 +0xf3
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:434 +0x22b
```

I think I see the bug:

https://github.com/cockroachdb/cockroach/blob/b2d5144601d8b7422200304fb9cd1ce426a0546f/pkg/sql/compact_sql_stats.go#L59

We are not checking the error before attempting to set the status.

--- a/pkg/sql/compact_sql_stats.go
+++ b/pkg/sql/compact_sql_stats.go
@@ -57,6 +57,9 @@ func (r *sqlStatsCompactionResumer) Resume(ctx context.Context, execCtx interfac
 
 		if scheduledJobID != jobs.InvalidScheduleID {
 			r.sj, err = jobs.LoadScheduledJob(ctx, scheduledjobs.ProdJobSchedulerEnv, scheduledJobID, ie, txn)
+			if err != nil {
+				return err
+			}
 			r.sj.SetScheduleStatus(string(jobs.StatusRunning))
 			return r.sj.Update(ctx, ie, txn)
 		}

